solr/core/src/java/org/apache/solr/core/backup/repository/LocalFileSystemRepository.java

```diff
@@ -97,7 +100,7 @@ public URI resolve(URI baseUri, String... pathComponents) {
   public void createDirectory(URI path) throws IOException {
     Path p = Path.of(path);
     if (!Files.exists(p, LinkOption.NOFOLLOW_LINKS)) {
-      Files.createDirectory(p);
```

Comment:
The new LocalFileSystemRepositoryTest revealed a small bug here :)
The method contract is to create all the directories needed, including the parents.